src/pmdas/bpf/modules/execsnoop.c

```diff
@@ -231,13 +231,13 @@ static void execsnoop_register(unsigned int cluster_id, pmdaMetric *metrics, pmd
             .units = PMDA_PMUNITS(0, 0, 0, 0, 0, 0),
         }
     };
-    /* bpf.execsnoop.uid */
+    /* bpf.execsnoop.lost */
     metrics[LOST] = (struct pmdaMetric)
     {
         .m_desc = {
             .pmid  = PMDA_PMID(cluster_id, 6),
             .type  = PM_TYPE_U32,
-            .indom = indom_id_mapping[EXECSNOOP_INDOM],
+            .indom = indom_id_mapping[LOST_EVENTS],
```

Comment:
I tried that and this break the bpf.execsnoop.lost metric 
```
$ git diff
diff --git a/src/pmdas/bpf/modules/execsnoop.c b/src/pmdas/bpf/modules/execsnoop.c
index 8b1b88034..e9239b6a5 100644
--- a/src/pmdas/bpf/modules/execsnoop.c
+++ b/src/pmdas/bpf/modules/execsnoop.c
@@ -237,7 +237,7 @@ static void execsnoop_register(unsigned int cluster_id, pmdaMetric *metrics, pmd
         .m_desc = {
             .pmid  = PMDA_PMID(cluster_id, 6),
             .type  = PM_TYPE_U32,
-            .indom = indom_id_mapping[LOST_EVENTS],
+            .indom = PM_INDOM_NULL,
             .sem   = PM_SEM_INSTANT,
             .units = PMDA_PMUNITS(0, 0, 0, 0, 0, 0),
         }
@@ -254,7 +254,7 @@ static void execsnoop_register(unsigned int cluster_id, pmdaMetric *metrics, pmd
     /* LOST_EVENTS InDom */
     indoms[1] = (struct pmdaIndom)
     {
-        indom_id_mapping[LOST_EVENTS],
+        PM_INDOM_NULL,
         1,
         execsnoop_lost,
     };

```

Reply:
```
--- a/src/pmdas/bpf/modules/execsnoop.c
+++ b/src/pmdas/bpf/modules/execsnoop.c
@@ -237,7 +237,7 @@ static void execsnoop_register(unsigned int cluster_id, pmdaMetric *metrics, pmd
         .m_desc = {
             .pmid  = PMDA_PMID(cluster_id, 6),
             .type  = PM_TYPE_U32,
-            .indom = indom_id_mapping[LOST_EVENTS],
+            .indom = PM_INDOM_NULL,
             .sem   = PM_SEM_INSTANT,
             .units = PMDA_PMUNITS(0, 0, 0, 0, 0, 0),
         }
```
:+1:

You don't need the `indoms[1]` declaration anymore. And in `execsnoop_fetch_to_atom` currently you have
```
    if (inst == PM_IN_NULL) {
        return PM_ERR_INST;
    }
```
(i.e., return if the `inst` is `PM_IN_NULL`), but in case of the lost metric without indom, inst will be `PM_IN_NULL`. You can move
```
    /* bpf.execsnoop.lost */
    if (item == LOST) {
        atom->ul = lost_events;
        return PMDA_FETCH_STATIC;
    }
```
up before this check for example, then it will work.

Reply:
Done. thanks. 
I applied this change to all other tools. 
however, still I got many instances from `bpf.bashreadline.lost` for example!